### DISCUSSION TITLE
CPU in percentage

### DISCUSSION THREAD

**Discussion by clumbo** [vectordotdev/vector#17178] [Q&A]
Hi,

I want to report the cpu in percent of each core every 10 secs, however I 'm not seeing a way to get the cpu in perecentage?

Am I misunderstanding the documents

Thanks

**Comment by bruceg** [comment]
There's not a direct way of getting percentages from the CPU metrics at this point. The emitted metrics are counters, which contain the cumulative number of CPU seconds since system start. Getting percentages would require computing an offset from a previous value and dividing by the time interval. Most metric analytic software should be able to do this for you from the cumulative counters.

**Reply by clumbo** [reply]
Thanks,

Would getting the the host up time work 

i.e cpu_seconds * 100 / up_time?

{"name":"uptime","namespace":"host","tags":{"collector":"host","host":"localhost"},"timestamp":"2023-04-19T20:10:06.746036026Z","kind":"absolute","gauge":{"value":112985.77}}

{"name":"cpu_seconds_total","namespace":"host","tags":{"collector":"cpu","cpu":"0","host":"localhost","mode":"idle"},"timestamp":"2023-04-19T20:10:06.746036026Z",
"kind":"absolute","counter":{"value":107942.95}}

**Reply by bruceg** [reply]
It would only work to produce an average CPU usage percentage over the entire uptime of the host. What I think you are asking for would require computing deltas for both of those values per desired reporting interval.

Technically, the delta of the uptime should be equal to the reporting interval, but it can vary slightly depending on internal delays.
